src/util.jl

```diff
@@ -39,22 +39,22 @@ end
 @pure tuple_minimum(T::Tuple) = minimum(T)
 
 # Something doesn't match up type wise
-function check_array_parameters(Size, T, N, L)
-    (!isa(Size, DataType) || (Size.name !== Tuple.name)) && throw(ArgumentError("Static Array parameter Size must be a Tuple type, got $Size"))
-    !isa(T, Type) && throw(ArgumentError("Static Array parameter T must be a type, got $T"))
-    !isa(N.parameters[1], Int) && throw(ArgumenError("Static Array parameter N must be an integer, got $(N.parameters[1])"))
-    !isa(L.parameters[1], Int) && throw(ArgumentError("Static Array parameter L must be an integer, got $(L.parameters[1])"))
+@generated function check_array_parameters(Size, T, N, L)
```

Comment:
I don't think this needs to be `@generated`, because this method only exists as a fallback signature for when the valid `check_array_parameters` signature further down fails to match.  So codegen does not need to be good for this one.

It took me five minutes to figure out why we even have two versions of `check_array_parameters` though.  Would be good to add a comment along the lines of

```
# This version of check_array_parameters() is a fallback signature to catch cases when invalid type
# parameters are passed.  It exists only for improved error reporting and should not be called by
# correct code.
```